### PR TITLE
Handle situations when bulk response has no items

### LINKED ISSUE #532: support absense of items key/value in the bulk response
in some situations it can happen that the _bulk index response doesn't contain the `items: []` element in it.
we should tolerate its absense since right now it causes an exception.

### PR DESCRIPTION
**Pull request by suyograo**
Fixes #532

### CODE REVIEW

File: lib/logstash/outputs/elasticsearch/http_client.rb [hunk 1]
```diff
@@ -91,7 +91,7 @@ def bulk(actions)
     def join_bulk_responses(bulk_responses)
       {
         "errors" => bulk_responses.any? {|r| r["errors"] == true},
-        "items" => bulk_responses.reduce([]) {|m,r| m.concat(r["items"])}
+        "items" => bulk_responses.reduce([]) {|m,r| m.concat(r.key?("items")? r["items"]: [])}
```
Comment (jsvd) [inline lib/logstash/outputs/elasticsearch/http_client.rb]: I suggest instead:
```ruby
"items" => bulk_responses.reduce([]) {|m,r| m.concat(r.fetch("items", []))}
```

File: spec/unit/outputs/elasticsearch/http_client_spec.rb [hunk 2]
```diff
@@ -97,6 +97,49 @@
     end
   end
 
+  describe "bulk response" do
```
Comment (jsvd) [inline spec/unit/outputs/elasticsearch/http_client_spec.rb]: I suggest `describe "join_bulk_responses" do` instead, this way we're describing the behaviour of this method.

**Comment by suyograo** [comment]
@jsvd updated.

**Comment by jsvd** [comment]
LGTM

**Pull request merged**